XamarinCommunityToolkit/Behaviors/TagBehavior.shared.cs

```diff
@@ -0,0 +1,157 @@
+
+                var lastIndex = 0;
+
+                foreach (Match item in collection)
+                {
+                    var text = textValue.Substring(lastIndex, item.Index - lastIndex);
+                    formatted.Spans.Add(CreateSpan(text));
+                    lastIndex = item.Index + item.Length;
+
+                    var span = CreateSpan(item.Value, true);
+
+                    formatted.Spans.Add(span);
+
+                    if (Command != null)
+                    {
+                        AddGestureRecognizer(span);
+                    }
+
+                }
+
+                var remainingText = textValue.Substring(lastIndex);
+
+                formatted.Spans.Add(CreateSpan(remainingText));
+            }
+        }
+
+        protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            base.OnPropertyChanged(propertyName);
+
+            if (propertyName == CommandProperty.PropertyName && Command != null)
+            {
+                ConfigureTags(View, AddGestureRecognizer);
+            }
+        }
+
+        protected override void OnDetachingFrom(Label bindable)
+        {
+            base.OnDetachingFrom(bindable);
+
+            ConfigureTags(bindable, RemoveGestureRecognizer);
+        }
+
+        void ConfigureTags(Label label, Action<Span> configAction)
+        {
+            if (label.FormattedText != null && label.FormattedText.Spans.Any())
+            {
+                var tagSpans = label.FormattedText.Spans.Where(p => Regex.Match(p.Text, RegexPattern).Success);
+                foreach (var span in tagSpans)
+                {
+                    configAction(span);
+                }
+            }
+        }
+
+        Span CreateSpan(string text, bool isTag = false)
+        {
+            var span = new Span()
+            {
+                Text = text,
+            };
+
+            if (isTag)
+            {
+                span.Style = TagTypes.FirstOrDefault(c => text.Contains(c.Symbol)).Style;
+            }
+
+            return span;
+        }
+
+        void AddGestureRecognizer(Span span)
+        {
+            var tapRecognizer = span.GestureRecognizers.FirstOrDefault() as TapGestureRecognizer;
+            if (tapRecognizer != null)
+            {
+                tapRecognizer.Command = Command;
+            }
+            else
+            {
+                span.GestureRecognizers.Add(new TapGestureRecognizer()
+                {
+                    Command = Command,
+                    CommandParameter = span.Text
+                });
+            }
+        }
+
+        void RemoveGestureRecognizer(Span span)
+        {
+            var tapRecognizer = span.GestureRecognizers.FirstOrDefault() as TapGestureRecognizer;
+            if (tapRecognizer != null)
+            {
+                span.GestureRecognizers.Remove(tapRecognizer);
+            }
+        }
+    }
+
+    public class TagType
```

Comment:
I believe, this class should be moved to the own file.

Reply:
Moved